pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -175,13 +175,18 @@ func (ghs *githubScraper) getOldestBranchCommit(repo *Repo, branch *Branch) {
 		ghs.logger.Sugar().Errorf("Error getting oldest commit", zap.Error(err))
 	}
 
-	oldestCommit, err := time.Parse(time.RFC3339, query.Repository.Ref.Target.Commit.History.Edges[0].Node.CommittedDate)
+    if len(query.Repository.Ref.Target.Commit.History.Edges) > 0 {
+        oldestCommit, err := time.Parse(time.RFC3339, query.Repository.Ref.Target.Commit.History.Edges[0].Node.CommittedDate)
 
-	if err != nil {
-		ghs.logger.Sugar().Errorf("Error converting timestamp for oldest commit", zap.Error(err))
-	}
+        if err != nil {
+            ghs.logger.Sugar().Errorf("Error converting timestamp for oldest commit", zap.Error(err))
+        }
+
+        branch.CreatedDate = oldestCommit
+    } else {
+        branch.CreatedDate = time.Now()
```

Comment:
@gesparza3 / @pmpaulino / @amber-beasley-liatrio  - There's a panic here where when a branch has been created, but no direct commits are made to it, then there will be an index out of range panic causing the collector to reset. This is because the graphql query doesn't return any edges. 

What I've done is check to ensure there are edges. But the `else` clause here i'm not a huge fan of. Currently that causes it to be set to 0 in the front end (as depicted in the screenshot below).

Thoughts on the way to handle that? One could see when the commit diverged from the main branch, but that causes a bunch of extra logic that would have to be handled and compared. I think that's technically the "right" way though. 

<img width="855" alt="image" src="https://github.com/liatrio/liatrio-otel-collector/assets/25961386/48769178-3522-40bb-a229-76219be108fe">

Reply:
Per [this thread](https://liatrio.slack.com/archives/C04KGSFF407/p1692126769426329?thread_ts=1692115050.438189&cid=C04KGSFF407) the decision was tentatively made to zero out the value & ignore it at the visualization layer. This enables the ability to infer a new metric out of this, and allows us to ignore the 0 values in the average calculations, while maintaining trust that the collector is still pulling accurate data. A decision log & documentation will be updated to reflect this.